packages/flutter_tools/lib/src/build_system/targets/shader_compiler.dart

```diff
@@ -175,6 +176,8 @@ class ShaderCompiler {
       );
     }
 
+    final String shaderLibPath = join(dirname(impellerc.path), 'shader_lib');
```

Comment:
You should use `_fs.path` here, that will work correctly if we are using a memory file system with a different style that the current host.

Reply:
Hmm could you expand on this? `_fs.path` seems to be the user project dir.

Reply:
Sorry, what I mean is using the `path` object defined on the local filesystem reference:

```dart
final String shaderLibPath = _fs.path.join(_fs.path.dirname(impellerc.path), 'shader_lib');
```

Reply:
Oh! Makes sense. Done.